src/CMakeLists.txt

```diff
@@ -392,36 +392,39 @@ if (ENABLE_BLACKMAGIC)
 endif()
 
 ################## OPENCV ###################
-
-find_package( OpenCV 4 )
-if (OpenCV_FOUND)
-  message("\nCOMPILING WITH OPENCV\n")
-  list(APPEND CMAKE_SWIG_FLAGS -DUSE_OPENCV=1)
-  target_compile_definitions(openshot PUBLIC USE_OPENCV=1)
-else()
-  message("\nOPENCV NOT FOUND, SOME FUNCTIONALITIES WILL BE DISABLED\n")
-endif()
-
-################## PROTOBUF ##################
-if (OpenCV_FOUND)
-  find_package(Protobuf 3)
-  
-  if (NOT Protobuf_FOUND)
-    # Protobuf is required when compiling with opencv
-    message(FATAL_ERROR "\nPLEASE INSTALL PROTOBUF. Protobuf is required when compiling with opencv.\n")
+if(ENABLE_OPENCV)
+  find_package(OpenCV 4)
+  if(NOT OpenCV_FOUND)
+    set(ENABLE_OPENCV FALSE CACHE BOOL
+      "Build with OpenCV algorithms (requires Boost, Protobuf 3)" FORCE)
+  else()
+    find_package(Protobuf 3 REQUIRED)
+    find_package(Boost REQUIRED)
+    set(PROTOBUF_SOURCE_FILES
+      protobuf_messages/objdetectdata.proto
+      protobuf_messages/stabilizedata.proto
+      protobuf_messages/trackerdata.proto
+    )
+    protobuf_generate_cpp(ProtoSources ProtoHeaders ${PROTOBUF_SOURCE_FILES})
+    # Add OpenCV source files and generated protobuf sources
+    target_sources(openshot PRIVATE
+      ${OPENSHOT_CV_SOURCES}
+      ${ProtoSources}
+      ${ProtoHeaders}
+    )
+    list(APPEND CMAKE_SWIG_FLAGS -DUSE_OPENCV=1)
+    target_compile_definitions(openshot PUBLIC USE_OPENCV=1)
+    target_link_libraries(openshot PUBLIC
+      opencv_core
+      opencv_video
+      opencv_highgui
+      opencv_dnn
+      opencv_tracking
+      protobuf::libprotobuf
```

Comment:
Boost isn't represented here at all, BTW, because it doesn't seem to be needed. If it's required for OpenCV and/or Protobuf, then presumably their target(s) take care of loading it automatically, and we should let them. But it's not required for any of our code AFAICT, is it?

Reply:
(I forgot to qualify, though, that if it should turn out that it's possible to end up with a system where OpenCV/Protobuf are installed, but the correct Boost components _aren't_, then this whole question may have to be revisited and we may need to do at least some detection/configuration for ourselves. But IMHO that's one of those "don't dick around solving it before you know if it's even possible to _have_ it" type of schrödingproblems.)